code/game/objects/items/candle.dm

```diff
@@ -63,7 +64,10 @@
 		return PROCESS_KILL
 	if(!infinite)
 		wax--
-	if(!wax)
+	if(explosive && !wax)
+		explosion(src.loc,0,2,3,flame_range = 3)
+		qdel(src)
+	if(!wax && !explosive)
 		new /obj/item/trash/candle(loc)
 		qdel(src)
```

Comment:
```suggestion
	if(!wax)
		if(explosive)
			explosion(src.loc,0,2,3,flame_range = 3)
			qdel(src)
		else
			new /obj/item/trash/candle(loc)
			qdel(src)
```

Reply:
tested and works just fine like this. thanks.